packages/odyssey-react-mui/src/labs/FileUploadPreview.tsx

```diff
@@ -0,0 +1,162 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { memo, useCallback } from "react";
+import styled from "@emotion/styled";
+import { IconButton } from "@mui/material";
+
+import { FileUploadProps } from "./FileUpload";
+import {
+  useOdysseyDesignTokens,
+  DesignTokens,
+} from "../OdysseyDesignTokensContext";
+import { DeleteIcon } from "../icons.generated";
+import { MuiPropsContext } from "../MuiPropsContext";
+import { Tooltip } from "../Tooltip";
+
+const PreviewContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  isDisabled: FileUploadProps["isDisabled"];
+  odysseyDesignTokens: DesignTokens;
+}>`
+  margin-block-start: ${({ odysseyDesignTokens }) =>
+    odysseyDesignTokens.Spacing2};
+  pointer-events: ${({ isDisabled }) => (isDisabled ? "none" : "normal")};
+  color: ${({ isDisabled, odysseyDesignTokens }) =>
+    isDisabled ? odysseyDesignTokens.TypographyColorDisabled : "inherit"};
+`;
+
+const UploadedFileContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: ${({ odysseyDesignTokens }) =>
+    `${odysseyDesignTokens.Spacing1} ${odysseyDesignTokens.Spacing2}`};
+  border-radius: ${({ odysseyDesignTokens }) =>
+    odysseyDesignTokens.BorderRadiusMain};
+  transition: ${({ odysseyDesignTokens }) =>
+    `background-color ${odysseyDesignTokens.TransitionTimingMain}`};
+
+  button {
+    transform: scale(0);
+  }
+
+  &:hover,
+  &:focus-within,
+  &:focus {
+    button {
+      transform: scale(1);
+    }
+  }
+
+  &:hover,
+  &:focus-within {
+    background-color: ${({ odysseyDesignTokens }) =>
+      odysseyDesignTokens.HueNeutral100};
+  }
+
+  &:focus {
+    border-color: ${({ odysseyDesignTokens }) =>
+      odysseyDesignTokens.FocusOutlineColorPrimary};
+    box-shadow: ${({ odysseyDesignTokens }) =>
+      `0 0 0 2px ${odysseyDesignTokens.FocusOutlineColorPrimary}`};
+    outline: ${({ odysseyDesignTokens }) =>
+      `${odysseyDesignTokens.FocusOutlineWidthMain} ${odysseyDesignTokens.FocusOutlineStyle} transparent`};
+  }
+`;
+
+type UploadedFileProps = {
+  name: string;
+  handleFileRemoval?: (name: string) => void;
+};
+
+const UploadedFile = ({
+  name,
+  handleFileRemoval,
+  ...rest
```

Comment:
We can remove this `...rest` because nothing else is being passed. If they are, we should type them separately.